### PR TITLE
[development] removed CryLibrary

### PR DESCRIPTION
**Pull request by AMZN-ScottR**
- Replaced CrySystem module entry points to use the one `AZ::DynamicModuleHandle` expects
 - Replaced loading of CrySystem with `AZ::DynamicModuleHandle`
 - Removed all references to and deleted CryLibrary
 - Removed unused `CSystem::SetAffinity` function
 - Removed unused `m_gameDll` member in `CGameEngine`
 - Removed unused `g_bProfilerEnabled` flag

Signed-off-by: AMZN-ScottR <24445312+AMZN-ScottR@users.noreply.github.com>

### CODE REVIEW

File: Code/LauncherUnified/Launcher.cpp [hunk 1]
```diff
@@ -649,17 +509,10 @@ namespace O3DELauncher
...
-        if (crySystemLibrary->Load(false))
+        AZStd::unique_ptr<CrySystemModuleHandle> crySystemLibrary = CrySystemModuleHandle::Create();
+        if (crySystemLibrary->Load())
         {
-            CreateSystemInterface = crySystemLibrary->GetFunction<PFNCREATESYSTEMINTERFACE>("CreateSystemInterface");
-            if (CreateSystemInterface)
-            {
-                systemInitParams.pSystem = CreateSystemInterface(systemInitParams);
-            }
+            systemInitParams.pSystem = crySystemLibrary->CreateSystemInterface(systemInitParams);
```
Comment (lumberyard-employee-dm) [inline Code/LauncherUnified/Launcher.cpp:515]: What was wrong with this code?

It seems like we could just use the `DynmicModuleHandle` directly as it was being used before

File: Code/Legacy/CryCommon/CrySystemLoader.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+#include <AzCore/Module/DynamicModuleHandle.h>
+#include <CryCommon/ISystem.h>
+
+constexpr const char* CrySystemLibraryName = AZ_TRAIT_OS_DYNAMIC_LIBRARY_PREFIX  "CrySystem" AZ_TRAIT_OS_DYNAMIC_LIBRARY_EXTENSION;
+
+constexpr const char* CreateInterfaceFunctionName = "CreateSystemInterface";
+
+constexpr const char* InjectEnvironmentFunctionName = "InjectEnvironment";
+constexpr const char* DetachEnvironmentFunctionName = "DetachEnvironment";
```
Comment (lumberyard-employee-dm) [inline Code/Legacy/CryCommon/CrySystemLoader.cpp:20]: We could rename the 
```
extern "C" AZ_DLL_EXPORT void InjectEnvironment(void* env);
extern "C" AZ_DLL_EXPORT void DetachEnvironment();
```
functions in [ISystem.h](https://github.com/o3de/o3de/blob/development/Code/Legacy/CryCommon/ISystem.h#L1107-L1108) to `InitializeDynamicModule` and `UninitializeDynamicModule` respectively.
The defintions of those functions are in [platform_impl.cpp](https://github.com/o3de/o3de/blob/development/Code/Legacy/CryCommon/platform_impl.cpp#L105-L119).
So once it is renamed there it should work like any other module

That way the regular DynamicModuleHandle could just be used directly.

Comment (AMZN-ScottR) [reply Code/Legacy/CryCommon/CrySystemLoader.cpp:20]: Great call, I'll do that instead

**Review by bosnichd — APPROVED** [review]
Agree with lumberyard-employee-dm's suggestion to use DynamicModuleHandle directly.

**Review by amznestebanpapp — COMMENTED** [review]
LGTM, will wait for the change suggested by @lumberyard-employee-dm

**Review by amznestebanpapp — APPROVED** [review]

**Pull request merged**